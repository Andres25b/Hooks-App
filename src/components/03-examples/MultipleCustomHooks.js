import React from 'react';
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';
import './css/style.css';


export const MultipleCustomHooks = () => {

    const {counter, increment} = useCounter(1);
    const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/characters/${counter}`);
    const {name, birthday, img, occupation} = !!data && data[0];

    // console.log(data);

    return (
        <div className=" mt-4">
            <h1>BreakingBad Characters</h1>
            <hr/>

            {
                loading
                ?
                    (
                        <div className="alert alert-info text-center">
                            Loading...
                        </div>
                    )
                :
                    (
                        <div className="container">
                            <div className="row">
                                <div className="col-md-4 offset-md-4 ">
                                    <div className="card">
                                        <img src={img} className="card-img-top" alt="name"/>
                                        <div className="card-body">
                                            <h5 className="card-title">{name}</h5>
                                            <div className="row">
                                                <div className="col-md-6">
                                                <p className="card-text mb-0"><span className="font-weight-bold">Birthday:</span></p>
                                                </div>
                                                <div className="col-md-6">
                                                    <p className="card-text mb-0">{birthday}</p>
                                                </div>
                                            </div>
                                            <div className="row">
                                                <div className="col-md-6">
                                                    <p className="font-weight-bold mb-0">Occupation: </p>
                                                </div>
                                                <div className="col-md-6">
                                                    {
                                                        occupation.map(o => {
                                                            return(
                                                                <p key={o} className="card-text mb-0">{o}</p>
                                                            );
                                                        })
                                                    }
                                                </div>
                                            </div>
                                        </div>
                                        <button className="btn btn-primary btn-block" onClick={increment}>Siguiente character</button>
                                    </div>
                                </div>
                            </div>
                        </div>

                    )
            }

        </div>
    )
}
