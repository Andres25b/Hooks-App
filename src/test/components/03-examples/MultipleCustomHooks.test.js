import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas en <MultipleCustomHooks />', () => {
    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    });

    test('debe de mostrarse correctamente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar la informacion', () => {

        useFetch.mockReturnValue({
            data: [{
                name: 'Andres', 
                birthday: '20-05-1996', 
                img: 'https://www.google.com/url?sa=i&url=http%3A%2F%2Felcanonazo.com%2Fbreaking-bad-transmedia-mucha-quimica%2F&psig=AOvVaw11jNGrnRgYy729OzoDamlD&ust=1594503961072000&source=images&cd=vfe&ved=0CAIQjRxqFwoTCJCAoqPUw-oCFQAAAAAdAAAAABAD', 
                occupation: ['Ingeniero en Sistemas']
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper.find('.alert').exists()).toBe(false);
        
    });

});
