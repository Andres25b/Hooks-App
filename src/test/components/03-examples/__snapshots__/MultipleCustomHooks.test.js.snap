// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe de mostrarse correctamente 1`] = `
<div
  className=" mt-4"
>
  <h1>
    BreakingBad Characters
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
</div>
`;
